lib/libsmb2.c

```diff
@@ -526,6 +526,8 @@ free_c_data(struct smb2_context *smb2, struct connect_data *c_data)
         free(discard_const(c_data->share));
         free(discard_const(c_data->user));
         free(c_data);
+
+        if ( smb2->connect_data == c_data ) smb2->connect_data = NULL;  /* to prevent double-free in smb2_destroy_context */
```

Comment:
We always use curly braces even if the body of the if-statement is just a single statement/expression, so this should be

```
if (smb2->connect_data == c_data) {
    smb2->connect_data = NULL;
}
```